x-pack/plugins/security/public/plugin.tsx

```diff
@@ -0,0 +1,148 @@
+import { SecurityNavControlService } from './nav_control';
+import { AccountManagementPage } from './account_management';
+import { AuthenticationService, AuthenticationServiceSetup } from './authentication';
+import { ManagementService, UserAPIClient } from './management';
+
+export interface PluginSetupDependencies {
+  licensing: LicensingPluginSetup;
+  home?: HomePublicPluginSetup;
+  management?: ManagementSetup;
+}
+
+export interface PluginStartDependencies {
+  data: DataPublicPluginStart;
+  management?: ManagementStart;
+}
+
+export class SecurityPlugin
+  implements
+    Plugin<
+      SecurityPluginSetup,
+      SecurityPluginStart,
+      PluginSetupDependencies,
+      PluginStartDependencies
+    > {
+  private sessionTimeout!: ISessionTimeout;
+  private readonly navControlService = new SecurityNavControlService();
+  private readonly securityLicenseService = new SecurityLicenseService();
+  private readonly managementService = new ManagementService();
+  private authc!: AuthenticationServiceSetup;
+
+  public setup(
+    core: CoreSetup<PluginStartDependencies>,
+    { home, licensing, management }: PluginSetupDependencies
+  ) {
+    const { http, notifications, injectedMetadata } = core;
+    const { basePath, anonymousPaths } = http;
+    anonymousPaths.register('/login');
+    anonymousPaths.register('/logout');
+    anonymousPaths.register('/logged_out');
+
+    const tenant = `${injectedMetadata.getInjectedVar('session.tenant', '')}`;
+    const sessionExpired = new SessionExpired(basePath, tenant);
+    http.intercept(new UnauthorizedResponseHttpInterceptor(sessionExpired, anonymousPaths));
+    this.sessionTimeout = new SessionTimeout(notifications, sessionExpired, http, tenant);
+    http.intercept(new SessionTimeoutHttpInterceptor(this.sessionTimeout, anonymousPaths));
+
+    const { license } = this.securityLicenseService.setup({ license$: licensing.license$ });
+
+    this.authc = new AuthenticationService().setup({ http: core.http });
+
+    this.navControlService.setup({
+      securityLicense: license,
+      authc: this.authc,
+    });
+
+    if (management) {
+      this.managementService.setup({
+        license,
+        management,
+        authc: this.authc,
+        fatalErrors: core.fatalErrors,
+        getStartServices: core.getStartServices,
+      });
+    }
+
+    // Home is an optional plugin, and if it's disabled we shouldn't try to fill feature catalog.
+    if (home) {
+      home.featureCatalogue.register({
+        id: 'security',
+        title: i18n.translate('xpack.security.registerFeature.securitySettingsTitle', {
+          defaultMessage: 'Security Settings',
+        }),
+        description: i18n.translate('xpack.security.registerFeature.securitySettingsDescription', {
+          defaultMessage:
+            'Protect your data and easily manage who has access to what with users and roles.',
+        }),
+        icon: 'securityApp',
+        path: '/app/kibana#/management/security',
```

Comment:
Since this feature catalogue entry links to our management apps, we should probably only register this entry if both `home` **and** `management` are enabled.

Reply:
Ha, good point!